libnetdata/socket/security.c

```diff
@@ -211,19 +323,34 @@ void security_start_ssl(int selector) {
                 info("To use encryption it is necessary to set \"ssl certificate\" and \"ssl key\" in [web] !\n");
                 return;
             }
-
+#ifdef NETDATA_HTTPS_WITH_OPENSSL
             netdata_srv_ctx =  security_initialize_openssl_server();
+#endif
+
+#ifdef NETDATA_HTTPS_WITH_WOLFSSL
+            netdata_srv_ctx =  security_initialize_wolfssl_server();
+#endif
             break;
         }
         case NETDATA_SSL_CONTEXT_STREAMING: {
+#ifdef NETDATA_HTTPS_WITH_OPENSSL
             netdata_client_ctx = security_initialize_openssl_client();
+#endif
+#ifdef NETDATA_HTTPS_WITH_WOLFSSL
+            netdata_client_ctx = security_initialize_wolfssl_client();
+#endif
             //This is necessary for the stream, because it is working sometimes with nonblock socket.
-            //It returns the bitmask afte to change, there is not any description of errors in the documentation
+            //It returns the bitmask after to change, there is not any description of errors in the documentation
             SSL_CTX_set_mode(netdata_client_ctx, SSL_MODE_ENABLE_PARTIAL_WRITE |SSL_MODE_ACCEPT_MOVING_WRITE_BUFFER |SSL_MODE_AUTO_RETRY);
             break;
         }
         case NETDATA_SSL_CONTEXT_EXPORTING: {
+#ifdef NETDATA_HTTPS_WITH_OPENSSL
             netdata_exporting_ctx = security_initialize_openssl_client();
+#endif
+#ifdef NETDATA_HTTPS_WITH_WOLFSSL
+            netdata_client_ctx = security_initialize_wolfssl_client();
+#endif
```

Comment:
I don't see anything related to the exporting engine in your code except this part.

You wrote
> Compile wolfssl on a system that does not have exporters like graphana, kinesis, mongodb or pubsub, because they can be linked against OpenSSL

We don't have a `graphana` connector.

What happens if you try to use these connectors with wolfSSL?

Reply:
I did not need to change anything inside the exporting, because we are using functions that are also present on `wolfssl`,  in theory all exporters should work without problems, but when I brought for the first time the `OpenSSL` to Netdata, the stream did not work correctly, until I do small adjusts inside our functions. We should test all possibilities to be sure that we do not need to do small changes on exporting, like it was necessary to do on streaming when OpenSSL is used.